source/adapters/level_zero/device.cpp

```diff
@@ -867,6 +867,16 @@ ur_result_t urDeviceGetInfo(
     return ReturnValue(int32_t(ZeDeviceNumIndices));
   } break;
   case UR_DEVICE_INFO_GPU_EU_COUNT: {
+    ze_device_properties_t DeviceProp = {};
+    DeviceProp.stype = ZE_STRUCTURE_TYPE_DEVICE_PROPERTIES;
+    ze_eu_count_ext_t EuCountDesc = {};
+    EuCountDesc.stype = ZE_STRUCTURE_TYPE_EU_COUNT_EXT;
+    DeviceProp.pNext = (void *)&EuCountDesc;
+    ZE2UR_CALL(zeDeviceGetProperties, (ZeDevice, &DeviceProp));
+    if (DeviceProp.pNext &&
+        EuCountDesc.stype == ZE_STRUCTURE_TYPE_EU_COUNT_EXT) {
+      return ReturnValue(uint32_t{EuCountDesc.numTotalEUs});
```

Comment:
should we check that this value is >0? I am not sure the driver is required to set the pNext to nullptr, this might always return 0.

Reply:
You should also be checking the driver extensions for ZE_extension_eu_count. If the driver does not report it supports 1.0 of this extension, then this pNext is guaranteed to not be set.